### PR TITLE
Support paid editions in agent-pool-terraform

### LINKED ISSUE #33652: Support non-Community editions in the agent-pool-terraform module
## Applies To

`/docs/agents/deploy-agents-terraform/`

## Details

The `agent-pool-terraform` module's [userdata script](https://github.com/gravitational/teleport/blob/2087a2fda8d57fe1d01e27b26da331daf5de8b94/examples/agent-pool-terraform/userdata#L3) only supports installing Teleport Community Edition. Let's change this to support commercial editions of Teleport as well.

## How will we know this is resolved?

A user can deploy an agent on a Teleport Enterprise/Team cluster using the `deploy-agents-terraform.mdx` guide.

### PR DESCRIPTION
**Pull request by ptgott**
Closes #33652

Currently, the `agent-pool-terraform` module only supports installing Teleport Community Edition on agent nodes, which is not compatible with commercial editions of Teleport.

This change adds an input variable for the user's Teleport edition and uses that input
variable to supply the second parameter of the one-line install script,
which specifies the Teleport edition. This way, we do not need to branch
the user data script within the module.

This change also includes the following edits, which were required to
test the guide:

- Put the version number in quotes within `main.auto.tfvars`.
- Add the `terraform init` command.
- Add an AMI data source instead of hardcoding an AMI ID. This prevents
  the AMI ID from going out of date.

For neatness, this also adds a separate `.tf` file for the token,
preventing the file from appearing too long when we include it in the
guide.

### CODE REVIEW

File: examples/agent-pool-terraform/userdata-ent [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# this fetches the correct package version.
+VERSION_ID=$(echo $VERSION_ID | grep -Eo "^[0-9]+")
+
+# Install the config-manager dnf plugin
+dnf install -y 'dnf-command(config-manager)'
+# Use the dnf config manager plugin to add the Teleport RPM repo
+dnf config-manager --add-repo "$(rpm --eval "https://yum.releases.teleport.dev/$ID/$VERSION_ID/Teleport/%_arch/stable/v${teleport_version}/teleport.repo")"
+
+# Install teleport
+dnf install -y teleport-ent
```
Comment (r0mant) [inline examples/agent-pool-terraform/userdata-ent:16]: So this will work only on RPM-based systems, which will be strange if an enterprise script would only support one type of systems while the OSS one had a proper install script.

Wouldn't a better way be updating our `https://goteleport.com/static/install.sh` install script to support passing an edition to it to take advantage of all its existing install logic?

Comment (ptgott) [reply examples/agent-pool-terraform/userdata-ent:16]: The Terraform module deploys an Amazon Linux 2023 instance, so these commands are meant to support that instance. The guide also [explains](https://github.com/gravitational/teleport/blob/2bab381f93ae453c9e98d08c9f121b60fde58a88/docs/pages/agents/deploy-agents-terraform.mdx#L151-L155) the need to edit the user data script for other environments.

I agree that it would be better to take advantage of the OS branching logic in `install.sh`, though. I'll look into getting the script edited.

**Comment by ptgott** [comment]
Hi @r0mant, I am following up to see if the recent edits to this PR address your feedback. Thanks!

File: examples/agent-pool-terraform/token.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+resource "teleport_provision_token" "agent" {
+  count = var.agent_count
+  spec = {
+    roles = [
+      "App",
+      "Db",
+      "Discovery",
+      "Kube",
+      "Node",
+    ]
```
Comment (hugoShaka) [inline examples/agent-pool-terraform/token.tf:15]: This is not directly related to the change, but it seems you're adding new roles. Could you make the roles a variable and tell the user to set only the needed ones? Granting all roles by default is a security anti-pattern.

File: docs/pages/agents/deploy-agents-terraform.mdx [hunk 3]
```diff
@@ -61,68 +61,95 @@ a demo cluster using:
...
-If you don't have an identify file available, make sure you have followed the
-[prerequisites for this guide](#prerequisites).
+1. Move the identity file for the Teleport Terraform provider into your project
+   directory so the Terraform provider an access it. Name the file
+   `terraform-identity`.
 
-</Notice>
+   <Notice type="warning">
+   
+   If you don't have an identify file available, make sure you have followed the
```
Comment (webvictim) [inline docs/pages/agents/deploy-agents-terraform.mdx:99]:
```suggestion
   If you don't have an identity file available, make sure you have followed the
```

**Review by webvictim — APPROVED** [review]

**Review by hugoShaka — APPROVED** [review]

**Review by r0mant — APPROVED** [review]
